File: atcoder.jp/contests/abc143/tasks/abc143_c/main.py
from itertools import groupby
from typing import List, Tuple


def run_length_encoding(s: str) -> List[Tuple[str, int]]:
    """ランレングス圧縮"""
    return [(k, len(list(g))) for k, g in groupby(s)]


N = int(input())
S = input()

rl = run_length_encoding(S)
ans = len(rl)

print(ans)

File: atcoder.jp/contests/abc143/tasks/abc143_c/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const runLengthEncoding = (str: string): [string, number][] => {
  const ret: [string, number][] = [[str[0], 1]];
  for (const c of str.slice(1)) {
    if (c === ret[ret.length - 1][0]) ret[ret.length - 1][1]++;
    else ret.push([c, 1]);
  }
  return ret;
};

const _N = +input();
const S = input();

const ans = runLengthEncoding(S).length;

console.log(ans);
